### PR TITLE
Spaceships example: bullet collisions and predict missing inputs

### PR DESCRIPTION
**Pull request by RJ**
* Bullet collisions added: collision events used for visuals on client, manipulating score on server.
* Predict missing remote player inputs

### CODE REVIEW

File: examples/spaceships/src/client.rs [hunk 1]
```diff
@@ -146,13 +148,65 @@ fn handle_new_player(
...
+                    &ActionState::default(),
+                    staleness,
+                    &mut aiq,
+                    tick,
+                );
+            } else {
+                apply_action_state_to_player_movement(prev_input, staleness, &mut aiq, tick);
+            }
+        } else {
+            apply_action_state_to_player_movement(action_state, 0, &mut aiq, tick);
```
Comment (cBournhonesque) [inline examples/spaceships/src/client.rs:209]: why do we have this here? we just apply the latest received ActionState?

**Review by cBournhonesque — APPROVED** [review]
LGTM!
The bloom and explosions look really good
I might have to move this from a `examples` folder to a `demo` folder as it is becoming quite big :)

**Comment by RJ** [comment]
> LGTM! The bloom and explosions look really good I might have to move this from a `examples` folder to a `demo` folder as it is becoming quite big :)

i convinced myself that the additional visuals are either instructive, or will help to debug network issues..

File: examples/spaceships/src/client.rs (hunk 1 above)
Comment (cBournhonesque) [reply examples/spaceships/src/client.rs:209]: I meant l209. Thanks, the comments are helpful

**Pull request merged**